<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Add correlation of ChIP-seq coverage to motif pairs. — addCor • sevenC</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Add correlation of ChIP-seq coverage to motif pairs. — addCor"><meta property="og:description" content="This function first adds ChIP-seq signals along all regions of motif location
using the function addCovToGR. Than it calculates the
correlation of coverage for each input pair using the function
addCovCor. The Pearson correlation coefficient is added as new
metadata column to the input interactions. Note, this function does not work
on windows because reading of bigWig files is currently not supported on
windows."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">sevenC</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.31.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/sevenC.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/ibn-salem/sevenC" class="external-link">
    <span class="fa fa-github"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Add correlation of ChIP-seq coverage to motif pairs.</h1>
    <small class="dont-index">Source: <a href="https://github.com/ibn-salem/sevenC/blob/devel/R/interaction.R" class="external-link"><code>R/interaction.R</code></a></small>
    <div class="hidden name"><code>addCor.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function first adds ChIP-seq signals along all regions of motif location
using the function <code><a href="addCovToGR.html">addCovToGR</a></code>. Than it calculates the
correlation of coverage for each input pair using the function
<code><a href="addCovCor.html">addCovCor</a></code>. The Pearson correlation coefficient is added as new
metadata column to the input interactions. Note, this function does not work
on windows because reading of bigWig files is currently not supported on
windows.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">addCor</span><span class="op">(</span><span class="va">gi</span>, <span class="va">bwFile</span>, name <span class="op">=</span> <span class="st">"chip"</span>, window <span class="op">=</span> <span class="fl">1000</span>, binSize <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-gi">gi<a class="anchor" aria-label="anchor" href="#arg-gi"></a></dt>
<dd><p><code><a href="https://rdrr.io/pkg/InteractionSet/man/GInteractions-class.html" class="external-link">GInteractions</a></code> object.</p></dd>


<dt id="arg-bwfile">bwFile<a class="anchor" aria-label="anchor" href="#arg-bwfile"></a></dt>
<dd><p>File path or connection to BigWig file with ChIP-seq signals.</p></dd>


<dt id="arg-name">name<a class="anchor" aria-label="anchor" href="#arg-name"></a></dt>
<dd><p>Character indicating the sample name.</p></dd>


<dt id="arg-window">window<a class="anchor" aria-label="anchor" href="#arg-window"></a></dt>
<dd><p>Numeric scalar for window size around the center of ranges in
<code>gr</code>.</p></dd>


<dt id="arg-binsize">binSize<a class="anchor" aria-label="anchor" href="#arg-binsize"></a></dt>
<dd><p>Integer scalar as size of bins to which the coverage values are
combined.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>An <code><a href="https://rdrr.io/pkg/InteractionSet/man/InteractionSet-class.html" class="external-link">GInteractions</a></code> object like <code>gi</code>
 with a new metadata column <code>colname</code> holding Pearson correlation
 coefficient of ChIP-seq signals for each anchor pair.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="va">.Platform</span><span class="op">$</span><span class="va">OS.type</span> <span class="op">!=</span> <span class="st">"windows"</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span> <span class="co"># use example bigWig file of ChIP-seq signals on human chromosome 22</span></span></span>
<span class="r-in"><span> <span class="va">exampleBigWig</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>,</span></span>
<span class="r-in"><span> <span class="st">"GM12878_Stat1.chr22_1-30000000.bigWig"</span>, package <span class="op">=</span> <span class="st">"sevenC"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span> <span class="co"># use example CTCF moitf location on human chromosome 22</span></span></span>
<span class="r-in"><span> <span class="va">motifGR</span> <span class="op">&lt;-</span> <span class="fu">sevenC</span><span class="fu">::</span><span class="va"><a href="motif.hg19.CTCF.chr22.html">motif.hg19.CTCF.chr22</a></span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span> <span class="co"># build candidate interactions</span></span></span>
<span class="r-in"><span> <span class="va">gi</span> <span class="op">&lt;-</span> <span class="fu"><a href="prepareCisPairs.html">prepareCisPairs</a></span><span class="op">(</span><span class="va">motifGR</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span> <span class="co"># add ChIP-seq signals correlation</span></span></span>
<span class="r-in"><span> <span class="va">gi</span> <span class="op">&lt;-</span> <span class="fu">addCor</span><span class="op">(</span><span class="va">gi</span>, <span class="va">exampleBigWig</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span> <span class="co"># use an alternative metadata column name for ChIP-seq correlation</span></span></span>
<span class="r-in"><span> <span class="va">gi</span> <span class="op">&lt;-</span> <span class="fu">addCor</span><span class="op">(</span><span class="va">gi</span>, <span class="va">exampleBigWig</span>, name <span class="op">=</span> <span class="st">"Stat1"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span> <span class="co"># add ChIP-seq correlation for signals signals in windows of 500bp around</span></span></span>
<span class="r-in"><span> <span class="co"># motif centers</span></span></span>
<span class="r-in"><span> <span class="va">gi</span> <span class="op">&lt;-</span> <span class="fu">addCor</span><span class="op">(</span><span class="va">gi</span>, <span class="va">exampleBigWig</span>, window <span class="op">=</span> <span class="fl">500</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span> <span class="co"># add ChIP-seq correlation for signals in bins of 10 bp</span></span></span>
<span class="r-in"><span> <span class="va">gi</span> <span class="op">&lt;-</span> <span class="fu">addCor</span><span class="op">(</span><span class="va">gi</span>, <span class="va">exampleBigWig</span>, binSize <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required namespace: GenomeInfoDb</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Jonas Ibn-Salem.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.9000.</p>
</div>

      </footer></div>






  </body></html>

