<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Function reference â€¢ sevenC</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">


<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script>
<script src="../pkgdown.js"></script>
  
  
<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-index">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">sevenC</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/sevenC.html">Get Started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/ibn-salem/sevenC">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

      <div class="row">
  <div class="col-md-9">
    <div class="page-header">
      <h1>
        Reference
        <small>version&nbsp;0.0.0.9000</small>
      </h1>
    </div>

    <div class="contents">
      <table class="ref-index">

      <colgroup>
        <col class="alias" />
        <col class="title" />
      </colgroup>

      <tbody>
        <tr>
          <th colspan="2">
            <h2 id="section-interactions" class="hasAnchor"><a href="#section-interactions" class="anchor"></a>Interactions</h2>
            <p class="section-desc"></p>
          </th>
        </tr>
        <tr>
          <!--  -->
          <td>
            <p><code><a href="prepareCisPairs.html">prepareCisPairs</a></code> </p>
          </td>
          <td><p>Prepares motif pairs as <code><a href='http://www.rdocumentation.org/packages/InteractionSet/topics/GInteractions-class'>GInteractions</a></code> and add
genomic features.</p></td>
        </tr><tr>
          <!--  -->
          <td>
            <p><code><a href="getCisPairs.html">getCisPairs</a></code> </p>
          </td>
          <td><p>Build a <code>GInteractions</code> object with all pairs of input
<code><a href='http://www.rdocumentation.org/packages/GenomicRanges/topics/GRanges-class'>GRanges</a></code> within a given distance.</p></td>
        </tr><tr>
          <!--  -->
          <td>
            <p><code><a href="addStrandCombination.html">addStrandCombination</a></code> </p>
          </td>
          <td><p>Add combination of anchor strand orientation.</p></td>
        </tr><tr>
          <!--  -->
          <td>
            <p><code><a href="addMotifScore.html">addMotifScore</a></code> </p>
          </td>
          <td><p>Add motif score of anchors.</p></td>
        </tr><tr>
          <!--  -->
          <td>
            <p><code><a href="addInteractionSupport.html">addInteractionSupport</a></code> </p>
          </td>
          <td><p>Add column to <code>GInteractions</code> with overlap support.</p></td>
        </tr><tr>
          <!--  -->
          <td>
            <p><code><a href="addCor.html">addCor</a></code> </p>
          </td>
          <td><p>Add correlation of ChIP-seq coverage to motif pairs.</p></td>
        </tr><tr>
          <!--  -->
          <td>
            <p><code><a href="addCovCor.html">addCovCor</a></code> </p>
          </td>
          <td><p>Add correlation of anchor signals to pairs of close genomic regions.</p></td>
        </tr>
      </tbody><tbody>
        <tr>
          <th colspan="2">
            <h2 id="section-reading-genomic-data" class="hasAnchor"><a href="#section-reading-genomic-data" class="anchor"></a>Reading genomic data</h2>
            <p class="section-desc"></p>
          </th>
        </tr>
        <tr>
          <!--  -->
          <td>
            <p><code><a href="addCovToGR.html">addCovToGR</a></code> </p>
          </td>
          <td><p>Add coverage to regions in <code><a href='http://www.rdocumentation.org/packages/GenomicRanges/topics/GRanges-class'>GRanges</a></code> object.</p></td>
        </tr><tr>
          <!--  -->
          <td>
            <p><code><a href="parseLoopsRao.html">parseLoopsRao</a></code> </p>
          </td>
          <td><p>Parse chromatin loops from Rao et al. 2014 as strict
<code><a href='http://www.rdocumentation.org/packages/InteractionSet/topics/GInteractions-class'>GInteractions</a></code>.</p></td>
        </tr><tr>
          <!--  -->
          <td>
            <p><code><a href="parseLoopsTang2015.html">parseLoopsTang2015</a></code> </p>
          </td>
          <td><p>Parse chromatin interactions from Tang et al. 2015 as <code>GInteractions</code>.</p></td>
        </tr>
      </tbody><tbody>
        <tr>
          <th colspan="2">
            <h2 id="section-prediction" class="hasAnchor"><a href="#section-prediction" class="anchor"></a>Prediction</h2>
            <p class="section-desc"></p>
          </th>
        </tr>
        <tr>
          <!--  -->
          <td>
            <p><code><a href="predLoops.html">predLoops</a></code> </p>
          </td>
          <td><p>Predict looping interactions.</p></td>
        </tr>
      </tbody><tbody>
        <tr>
          <th colspan="2">
            <h2 id="section-data-sets" class="hasAnchor"><a href="#section-data-sets" class="anchor"></a>Data sets</h2>
            <p class="section-desc"></p>
          </th>
        </tr>
        <tr>
          <!--  -->
          <td>
            <p><code><a href="motif.hg19.CTCF.html">motif.hg19.CTCF</a></code> </p>
          </td>
          <td><p>CTCF motif locations in human genome hg19.</p></td>
        </tr><tr>
          <!--  -->
          <td>
            <p><code><a href="motif.hg19.CTCF.chr22.html">motif.hg19.CTCF.chr22</a></code> </p>
          </td>
          <td><p>CTCF motif locations on chromosome 22 in human genome hg19.</p></td>
        </tr><tr>
          <!--  -->
          <td>
            <p><code><a href="motif.hg19.CTCF.chr22.cov.html">motif.hg19.CTCF.chr22.cov</a></code> </p>
          </td>
          <td><p>CTCF motifs on human chromosome 22 with example coverage.</p></td>
        </tr><tr>
          <!--  -->
          <td>
            <p><code><a href="modelBest10Avg.html">modelBest10Avg</a></code> </p>
          </td>
          <td><p>Default parameters for logistic regression model.</p></td>
        </tr><tr>
          <!--  -->
          <td>
            <p><code><a href="cutoffByTF.html">cutoffByTF</a></code> </p>
          </td>
          <td><p>Optimal cutoff values for logistic regression models.</p></td>
        </tr><tr>
          <!--  -->
          <td>
            <p><code><a href="cutoffBest10.html">cutoffBest10</a></code> </p>
          </td>
          <td><p>Default optimal cutoff value of logistic regression.</p></td>
        </tr>
      </tbody>
      </table>
    </div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#section-interactions">Interactions</a></li>
      <li><a href="#section-reading-genomic-data">Reading genomic data</a></li>
      <li><a href="#section-prediction">Prediction</a></li>
      <li><a href="#section-data-sets">Data sets</a></li>
    </ul>
  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Jonas Ibn-Salem.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  </body>
</html>
