<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Prediction of chromatin looping interactions with *sevenC* • sevenC</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Prediction of chromatin looping interactions with *sevenC*">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">sevenC</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.31.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/sevenC.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/ibn-salem/sevenC" class="external-link">
    <span class="fa fa-github"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Prediction of chromatin looping interactions
with <em>sevenC</em>
</h1>
                        <h4 data-toc-skip class="author">Jonas
Ibn-Salem</h4>
            <address class="author_afil">
      Faculty of Biology, Johannes Gutenberg University of Mainz, 55128
Mainz, GermanyInstitute of Molecular Biology, 55128 Mainz,
Germany<br><a class="author_email" href="mailto:#"></a><a href="mailto:j.ibn-salem@uni-mainz.de" class="email">j.ibn-salem@uni-mainz.de</a>
      </address>
                              <h4 data-toc-skip class="author">Miguel
Andrade-Navarro</h4>
            <address class="author_afil">
      Faculty of Biology, Johannes Gutenberg University of Mainz, 55128
Mainz, GermanyInstitute of Molecular Biology, 55128 Mainz,
Germany<br><small class="dont-index">Source: <a href="https://github.com/ibn-salem/sevenC/blob/devel/vignettes/sevenC.Rmd" class="external-link"><code>vignettes/sevenC.Rmd</code></a></small>
      <div class="hidden name"><code>sevenC.Rmd</code></div>

    </address>
</div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>Chromatin looping is an essential feature of eukaryotic genomes
      and can bring regulatory sequences, such as enhancers or
      transcription factor binding sites, in the close physical
      proximity of regulated target genes. Here, we provide sevenC, an R
      package that uses protein binding signals from ChIP-seq and
      sequence motif information to predict chromatin looping events.
      Cross-linking of proteins that bind close to loop anchors result
      in ChIP-seq signals at both anchor loci. These signals are used at
      CTCF motif pairs together with their distance and orientation to
      each other to predict whether they interact or not. The resulting
      chromatin loops might be used to associate enhancers or
      transcription factor binding sites (e.g., ChIP-seq peaks) to
      regulated target genes.</p>
    </div>
    
<div class="section level2">
<h2 id="background-and-introduction">Background and introduction<a class="anchor" aria-label="anchor" href="#background-and-introduction"></a>
</h2>
<p>Gene expression is regulated by binding of transcription factors (TF)
to genomic DNA. However, many binding sites are in distal regulatory
regions, such as enhancers, that are hundreds of kilobases apart from
genes. These regulatory regions can physically interact with promoters
of regulated genes by chromatin looping interactions. These looping
interaction can be measured genome-wide by chromatin conformation
capture techniques such as Hi-C or ChIA-PET <span class="citation">(Rao
et al. 2014; Tang et al. 2015)</span>. Despite many exciting insights
into the three-dimensional organization of genomes, these experimental
methods are not only elaborate and expansive but also have limited
resolution and are only available for a limited number of cell types and
conditions. In contrast, the binding sites of TFs can be detected
genome-wide by ChIP-seq experiment with high resolution and are
available for hundreds of TFs in many cell type and conditions. However,
classical analysis of ChIP-seq gives only the direct binding sites of
targeted TFs (ChIP-seq peaks) and it is not trivial to associate them to
the regulated gene without chromatin looping information. Therefore, we
provide a computational method to predict chromatin interactions from
only genomic sequence features and ChIP-seq data. The predicted looping
interactions can be used to associate TF binding sites (ChIP-seq peaks)
or enhancers to regulated genes and thereby improve functional
downstream analysis on the level of genes.</p>
<p>In this vignette, we show how to use the R package <em><a href="https://bioconductor.org/packages/3.22/sevenC" class="external-link">sevenC</a></em> to
predict chromatin looping interactions between CTCF motifs by using only
ChIP-seq data form a single experiment. Furthermore, we show how to
train the prediction model using custom data.</p>
<p>A more detailed explanation of the sevenC method together with
prediction performance analysis is available in the associated preprint
<span class="citation">(Ibn-Salem and Andrade-Navarro 2018)</span>.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>To install the <em>sevenC</em> package, start R and enter:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install.packages("BiocManager")</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"sevenC"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="predict-chromatin-looping-interactions">Predict chromatin looping interactions<a class="anchor" aria-label="anchor" href="#predict-chromatin-looping-interactions"></a>
</h2>
<div class="section level3">
<h3 id="basic-usage-example">Basic usage example<a class="anchor" aria-label="anchor" href="#basic-usage-example"></a>
</h3>
<p>Here we show how to use the <em><a href="https://bioconductor.org/packages/3.22/sevenC" class="external-link">sevenC</a></em>
package with default options to predict chromatin looping interactions
among CTCF motif locations on the human chromosome 22. As input, we only
use CTCF motif locations and a single bigWig file from a STAT1 ChIP-seq
experiment in human GM12878 cells <span class="citation">(Dunham et al.
2012)</span>.</p>
<div class="section level4">
<h4 id="get-motif-pairs">Get motif pairs<a class="anchor" aria-label="anchor" href="#get-motif-pairs"></a>
</h4>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/ibn-salem/sevenC" class="external-link">sevenC</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># load provided CTCF motifs in human genome</span></span>
<span><span class="va">motifs</span> <span class="op">&lt;-</span> <span class="va">motif.hg19.CTCF.chr22</span></span>
<span></span>
<span><span class="co"># get motifs pairs</span></span>
<span><span class="va">gi</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prepareCisPairs.html">prepareCisPairs</a></span><span class="op">(</span><span class="va">motifs</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="add-chip-seq-data-and-compute-correaltion">Add ChIP-seq data and compute correaltion<a class="anchor" aria-label="anchor" href="#add-chip-seq-data-and-compute-correaltion"></a>
</h4>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># use example ChIP-seq bigWig file</span></span>
<span><span class="va">bigWigFile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"GM12878_Stat1.chr22_1-30000000.bigWig"</span>, </span>
<span>  package <span class="op">=</span> <span class="st">"sevenC"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># add ChIP-seq coverage and compute correaltion at motif pairs</span></span>
<span><span class="va">gi</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/addCor.html">addCor</a></span><span class="op">(</span><span class="va">gi</span>, <span class="va">bigWigFile</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="predict-loops">Predict loops<a class="anchor" aria-label="anchor" href="#predict-loops"></a>
</h4>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># predict looping interactions among all motif pairs</span></span>
<span><span class="va">loops</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/predLoops.html">predLoops</a></span><span class="op">(</span><span class="va">gi</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="more-detailed-usage-example">More detailed usage example<a class="anchor" aria-label="anchor" href="#more-detailed-usage-example"></a>
</h3>
<p>Here we show in more detail each step of the loop prediction process.
Again, we want to predict chromatin looping interactions among CTCF
motif locations on chromosome 22 using a ChIP-seq for STAT1 in human
GM12878 cells.</p>
<div class="section level4">
<h4 id="prepare-ctcf-motif-pairs">Prepare CTCF motif pairs<a class="anchor" aria-label="anchor" href="#prepare-ctcf-motif-pairs"></a>
</h4>
<p>First, we need to prepare CTCF motif pairs as candidate anchors for
chromatin loop interactions. We use CTCF motif hits in human chromosome
22 as provide by <em><a href="https://bioconductor.org/packages/3.22/sevenC" class="external-link">sevenC</a></em>
package. In general, any CTCF motifs can be used if provided as
<code>GRanges</code>. To use the motif similarity score as a predictive
feature, the motif data should contain -log<sub>10</sub> transformed
p-values describing the significance of each motif hit. Here, we use
CTCF motif sites as provided from the JASPAR genome browser tracks <span class="citation">(Khan et al. 2018)</span>. The objedt
<code>motif.hg19.CTCF.chr22</code> in the
<code>r BiocStyle::Biocpkg("sevenC")</code> package contains CTCF motif
locations on chromosome 22. For more information on the motif data set,
see <code><a href="../reference/motif.hg19.CTCF.html">?motif.hg19.CTCF</a></code>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/ibn-salem/sevenC" class="external-link">sevenC</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># load provided CTCF motifs</span></span>
<span><span class="va">motifs</span> <span class="op">&lt;-</span> <span class="va">motif.hg19.CTCF.chr22</span></span></code></pre></div>
<p>The CTCF motif are represented as <code>GRanges</code> object from
the <code>r BiocStyle::Biocpkg("GenomicRanges")</code> package. There
are 917 CTCF motif locations on chromosome 22. The genome assembly is
hg19. one metadata column named <code>score</code> shows motif match
similarity as -log<sub>10</sub> transformed p-value.</p>
</div>
<div class="section level4">
<h4 id="add-chip-seq-signals-at-motifs-sites">Add ChIP-seq signals at motifs sites<a class="anchor" aria-label="anchor" href="#add-chip-seq-signals-at-motifs-sites"></a>
</h4>
<p>To predict loops, we need the ChIP-seq signals at all motif sites.
Therefore, we read an example bigWig file with ChIP-seq signals.</p>
<p>An example file with only data on a subset of chromosome 22 is
provided as part of the <em><a href="https://bioconductor.org/packages/3.22/sevenC" class="external-link">sevenC</a></em>
package. The full file can be downloaded from ENCODE <span class="citation">(Dunham et al. 2012)</span> <a href="http://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeSydhTfbs/wgEncodeSydhTfbsGm12878Stat1StdSig.bigWig" class="external-link">here</a>.
The file contains for each position in the genome the log-fold-change of
ChIP-seq signals versus input control.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># use example ChIP-seq bigWig file</span></span>
<span><span class="va">bigWigFile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"GM12878_Stat1.chr22_1-30000000.bigWig"</span>, </span>
<span>  package <span class="op">=</span> <span class="st">"sevenC"</span><span class="op">)</span></span></code></pre></div>
<p>We add ChIP-seq signals to all motifs in a window of 1000 bp using
the function <code><a href="../reference/addCovToGR.html">addCovToGR()</a></code> as follows.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># read ChIP-seq coverage </span></span>
<span><span class="va">motifs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/addCovToGR.html">addCovToGR</a></span><span class="op">(</span><span class="va">motifs</span>, <span class="va">bigWigFile</span><span class="op">)</span></span></code></pre></div>
<p>This adds a new metadata column to <code>motifs</code> holding a
<code>NumericList</code> with ChIP-seq signals for each motif
location.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">motifs</span><span class="op">$</span><span class="va">chip</span></span></code></pre></div>
<pre><code><span><span class="co">## NumericList of length 917</span></span>
<span><span class="co">## [["chr22"]] 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 ... 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1</span></span>
<span><span class="co">## [["chr22"]] 1 1 1 1 1 0 0 0 0 0 0 0 0 0 0 0 ... 1 1 2 2 2 2 2 2 2 2 2 2 2 2 2 2</span></span>
<span><span class="co">## [["chr22"]] 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 ... 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0</span></span>
<span><span class="co">## [["chr22"]] 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 ... 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1</span></span>
<span><span class="co">## [["chr22"]] 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 ... 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0</span></span>
<span><span class="co">## [["chr22"]] 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 ... 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0</span></span>
<span><span class="co">## [["chr22"]] 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 ... 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0</span></span>
<span><span class="co">## [["chr22"]] 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 ... 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0</span></span>
<span><span class="co">## [["chr22"]] 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 ... 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0</span></span>
<span><span class="co">## [["chr22"]] 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 ... 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2</span></span>
<span><span class="co">## ...</span></span>
<span><span class="co">## &lt;907 more elements&gt;</span></span></code></pre>
<p>Please note, on Windows systems, reading of bigWig files is currently
not supported. See <code>help(rtracklayer::import.bw)</code> for more
information. Users on Windows need to get ChIP-seq signals around motif
sites as a <code>NumierList</code> object. A <code>NumericList</code>
<code>l</code> with ChIP-signal counts around each motif list can be
added by <code>motifs$chip &lt;- l</code>.</p>
</div>
<div class="section level4">
<h4 id="build-pairs-of-motifs-as-candidate-interactions">Build pairs of motifs as candidate interactions<a class="anchor" aria-label="anchor" href="#build-pairs-of-motifs-as-candidate-interactions"></a>
</h4>
<p>Now we build a dataset with all pairs of CTCF motif within 1 Mb and
annotate it with distance, motif orientation, and motif score.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gi</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prepareCisPairs.html">prepareCisPairs</a></span><span class="op">(</span><span class="va">motifs</span>, maxDist <span class="op">=</span> <span class="fl">10</span><span class="op">^</span><span class="fl">6</span><span class="op">)</span></span>
<span></span>
<span><span class="va">gi</span></span></code></pre></div>
<pre><code><span><span class="co">## StrictGInteractions object with 26076 interactions and 5 metadata columns:</span></span>
<span><span class="co">##           seqnames1           ranges1     seqnames2           ranges2 |</span></span>
<span><span class="co">##               &lt;Rle&gt;         &lt;IRanges&gt;         &lt;Rle&gt;         &lt;IRanges&gt; |</span></span>
<span><span class="co">##       [1]     chr22 16186188-16186206 ---     chr22 16205307-16205325 |</span></span>
<span><span class="co">##       [2]     chr22 16186188-16186206 ---     chr22 16238548-16238566 |</span></span>
<span><span class="co">##       [3]     chr22 16186188-16186206 ---     chr22 16239188-16239206 |</span></span>
<span><span class="co">##       [4]     chr22 16186188-16186206 ---     chr22 16239827-16239845 |</span></span>
<span><span class="co">##       [5]     chr22 16186188-16186206 ---     chr22 16247246-16247264 |</span></span>
<span><span class="co">##       ...       ...               ... ...       ...               ... .</span></span>
<span><span class="co">##   [26072]     chr22 51110780-51110798 ---     chr22 51160028-51160046 |</span></span>
<span><span class="co">##   [26073]     chr22 51110780-51110798 ---     chr22 51172084-51172102 |</span></span>
<span><span class="co">##   [26074]     chr22 51130130-51130148 ---     chr22 51160028-51160046 |</span></span>
<span><span class="co">##   [26075]     chr22 51130130-51130148 ---     chr22 51172084-51172102 |</span></span>
<span><span class="co">##   [26076]     chr22 51160028-51160046 ---     chr22 51172084-51172102 |</span></span>
<span><span class="co">##                dist strandOrientation   score_1   score_2 score_min</span></span>
<span><span class="co">##           &lt;integer&gt;       &lt;character&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">##       [1]     19119           forward      6.05      5.72      5.72</span></span>
<span><span class="co">##       [2]     52360           forward      6.05      5.92      5.92</span></span>
<span><span class="co">##       [3]     53000           forward      6.05      5.90      5.90</span></span>
<span><span class="co">##       [4]     53639           forward      6.05      5.92      5.92</span></span>
<span><span class="co">##       [5]     61058           forward      6.05      5.92      5.92</span></span>
<span><span class="co">##       ...       ...               ...       ...       ...       ...</span></span>
<span><span class="co">##   [26072]     49248           reverse      6.21      5.64      5.64</span></span>
<span><span class="co">##   [26073]     61304           reverse      6.21      5.90      5.90</span></span>
<span><span class="co">##   [26074]     29898           reverse      6.33      5.64      5.64</span></span>
<span><span class="co">##   [26075]     41954           reverse      6.33      5.90      5.90</span></span>
<span><span class="co">##   [26076]     12056           reverse      5.64      5.90      5.64</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   regions: 917 ranges and 2 metadata columns</span></span>
<span><span class="co">##   seqinfo: 93 sequences (1 circular) from hg19 genome</span></span></code></pre>
<p>The function <code><a href="../reference/prepareCisPairs.html">prepareCisPairs()</a></code> returns a
<code>GInteractoin</code> object from the
<code>r BiocStyle::Biocpkg("InteractonSet")</code> package, representing
all motif pairs within the defined distance. The metadata columns of the
<code>GInteractoin</code> object hold the genomic distance between
motifs in bp (<code>dist</code>), the orientation of motifs
(<code>strandOrientation</code>), and the motif score as
-log<sub>10</sub> of the motif hit p-value (<code>score_1</code>,
<code>score_2</code>, and <code>score_min</code>). Note, that the
function <code><a href="../reference/prepareCisPairs.html">prepareCisPairs()</a></code> is a wrapper for three
individual functions that perform each step separately and allow more
options. First, <code><a href="../reference/getCisPairs.html">getCisPairs()</a></code> is used to builds the
<code>GInteractoin</code> object. Than
<code><a href="../reference/addStrandCombination.html">addStrandCombination()</a></code> adds the four possible strand
combinations of motifs pairs. Finally, <code><a href="../reference/addMotifScore.html">addMotifScore()</a></code> adds
the minimum motif score for each pair. These genomic features are used
later as predictive variables.</p>
</div>
</div>
<div class="section level3">
<h3 id="compute-chip-seq-similarity-at-motif-pairs">Compute ChIP-seq similarity at motif pairs<a class="anchor" aria-label="anchor" href="#compute-chip-seq-similarity-at-motif-pairs"></a>
</h3>
<p>Now, we compute the similarity of ChIP-seq signals for all motif
pairs as the correlation of signals across positions around motif
centers. Thereby, for two motifs the corresponding ChIP-seq signal
vectors that were added to <code>motifs</code> before, are compared by
Pearson correlation. A high correlation of ChIP-seq signals at two
motifs indicates a similar ChIP-seq coverage profile at the two motifs.
This, in turn, is characteristic for physical interaction via chromatin
looping, where ChIP signals are found on both sides with a similar
distance to motif centers <span class="citation">(Ibn-Salem and
Andrade-Navarro 2018)</span>. The correlation coefficient is added as
additional metadata column to <code>gi</code>.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># add ChIP-seq coverage and compute correaltion at motif pairs</span></span>
<span><span class="va">gi</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/addCovCor.html">addCovCor</a></span><span class="op">(</span><span class="va">gi</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="predict-loops-1">Predict loops<a class="anchor" aria-label="anchor" href="#predict-loops-1"></a>
</h3>
<p>Now we can predict chromatin loops integrating from the ChIP-seq
correlation and other genomic features in a logistic regression model.
This is implemented in the <code><a href="../reference/predLoops.html">predLoops()</a></code> function.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">loops</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/predLoops.html">predLoops</a></span><span class="op">(</span><span class="va">gi</span><span class="op">)</span></span>
<span></span>
<span><span class="va">loops</span></span></code></pre></div>
<pre><code><span><span class="co">## StrictGInteractions object with 607 interactions and 7 metadata columns:</span></span>
<span><span class="co">##         seqnames1           ranges1     seqnames2           ranges2 |      dist</span></span>
<span><span class="co">##             &lt;Rle&gt;         &lt;IRanges&gt;         &lt;Rle&gt;         &lt;IRanges&gt; | &lt;integer&gt;</span></span>
<span><span class="co">##     [1]     chr22 16186188-16186206 ---     chr22 16239188-16239206 |     53000</span></span>
<span><span class="co">##     [2]     chr22 16205307-16205325 ---     chr22 16409786-16409804 |    204479</span></span>
<span><span class="co">##     [3]     chr22 16238548-16238566 ---     chr22 16409786-16409804 |    171238</span></span>
<span><span class="co">##     [4]     chr22 17398482-17398500 ---     chr22 17539262-17539280 |    140780</span></span>
<span><span class="co">##     [5]     chr22 17398482-17398500 ---     chr22 17652850-17652868 |    254368</span></span>
<span><span class="co">##     ...       ...               ... ...       ...               ... .       ...</span></span>
<span><span class="co">##   [603]     chr22 26756542-26756560 ---     chr22 27041422-27041440 |    284880</span></span>
<span><span class="co">##   [604]     chr22 26763796-26763814 ---     chr22 27041422-27041440 |    277626</span></span>
<span><span class="co">##   [605]     chr22 28123236-28123254 ---     chr22 28221328-28221346 |     98092</span></span>
<span><span class="co">##   [606]     chr22 29225543-29225561 ---     chr22 29436847-29436865 |    211304</span></span>
<span><span class="co">##   [607]     chr22 29436847-29436865 ---     chr22 29641698-29641716 |    204851</span></span>
<span><span class="co">##         strandOrientation   score_1   score_2 score_min  cor_chip      pred</span></span>
<span><span class="co">##               &lt;character&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">##     [1]           forward      6.05      5.90      5.90  0.885073  0.193089</span></span>
<span><span class="co">##     [2]        convergent      5.72      5.79      5.72  0.415731  0.162644</span></span>
<span><span class="co">##     [3]        convergent      5.92      5.79      5.79  0.675000  0.301512</span></span>
<span><span class="co">##     [4]        convergent      6.48      6.89      6.48  0.790284  0.550519</span></span>
<span><span class="co">##     [5]        convergent      6.48      8.49      6.48  0.221485  0.172598</span></span>
<span><span class="co">##     ...               ...       ...       ...       ...       ...       ...</span></span>
<span><span class="co">##   [603]           reverse      8.27      7.93      7.93  0.349559  0.225008</span></span>
<span><span class="co">##   [604]           reverse      8.59      7.93      7.93  0.739380  0.423643</span></span>
<span><span class="co">##   [605]           reverse      6.29      8.16      6.29  0.601548  0.188785</span></span>
<span><span class="co">##   [606]           reverse      6.73      6.33      6.33  0.928965  0.245978</span></span>
<span><span class="co">##   [607]           reverse      6.33      7.27      6.33  0.909619  0.242532</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   regions: 917 ranges and 2 metadata columns</span></span>
<span><span class="co">##   seqinfo: 93 sequences (1 circular) from hg19 genome</span></span></code></pre>
<p>The <code><a href="../reference/predLoops.html">predLoops()</a></code> function returns a subset of motif pairs
that are predicted to interact. The interactions are annotated with
ChIP-seq correlation in column <code>cor_chip</code>. The column
<code>pred</code> holds the predicted interaction probability according
to the logistic regression model.</p>
<p>Note, that without specifying further options, the function
<code><a href="../reference/predLoops.html">predLoops()</a></code> uses a default model that was optimized for
several transcription factor ChIP-seq datasets by using experimental
chromatin loops from Hi-C and ChIA-PET for validations <span class="citation">(Ibn-Salem and Andrade-Navarro 2018)</span>. However,
users can specify custom features using the <code>formula</code>
argument and provide custom parameters using the <code>betas</code>
argument. Furthermore, per default the <code><a href="../reference/predLoops.html">predLoops()</a></code> function
report only looping interactions that reach a minimal prediction score
threshold. The fraction of reported loops can be modified using the
<code>cutoff</code> argument.</p>
</div>
</div>
<div class="section level2">
<h2 id="downstream-analysis-with-predicted-chromatin-loops">Downstream analysis with predicted chromatin loops<a class="anchor" aria-label="anchor" href="#downstream-analysis-with-predicted-chromatin-loops"></a>
</h2>
<div class="section level3">
<h3 id="linking-sets-of-regions">Linking sets of regions<a class="anchor" aria-label="anchor" href="#linking-sets-of-regions"></a>
</h3>
<p>Predicted loops are represented as <code>GInteraction</code> and can,
therefore, be used easily for downstream analysis with functions from
the <code>r BiocStyle::Biocpkg("InteractonSet")</code> package. For
example, linking two sets of regions (like ChIP-seq peaks and genes) can
be done using the <code>linkOverlaps</code> function. See the <a href="http://bioconductor.org/packages/release/bioc/vignettes/InteractionSet/inst/doc/interactions.html" class="external-link">vignette</a>
from the <em><a href="https://bioconductor.org/packages/3.22/InteractonSet" class="external-link">InteractonSet</a></em>
package for more details and examples on working with
<code>GInteraction</code> objects.</p>
</div>
<div class="section level3">
<h3 id="write-predicted-loops-to-an-output-file">Write predicted loops to an output file<a class="anchor" aria-label="anchor" href="#write-predicted-loops-to-an-output-file"></a>
</h3>
<p>Since looping interactions are stored as <code>GInteraction</code>
objects, they can be exported as <a href="http://bedtools.readthedocs.io/en/latest/content/general-usage.html#bedpe-format" class="external-link">BEDPE</a>
files using functions from <em><a href="https://bioconductor.org/packages/3.22/GenomicInteractions" class="external-link">GenomicInteractions</a></em>
package. These files can be used for visualization in genome browsers or
the <a href="https://www.aidenlab.org/juicebox/" class="external-link">Juicebox</a> tool.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/ComputationalRegulatoryGenomicsICL/GenomicInteractions/" class="external-link">GenomicInteractions</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># export to output file</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GenomicInteractions/man/export.bedpe.html" class="external-link">export.bedpe</a></span><span class="op">(</span><span class="va">loops</span>, <span class="st">"loop_interactions.bedpe"</span>, score <span class="op">=</span> <span class="st">"pred"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="train-prediction-model-using-custom-data">Train prediction model using custom data<a class="anchor" aria-label="anchor" href="#train-prediction-model-using-custom-data"></a>
</h2>
<p>Here, we show how to use <em><a href="https://bioconductor.org/packages/3.22/sevenC" class="external-link">sevenC</a></em> to
build and train a logistic regression model for loop prediction.</p>
<div class="section level3">
<h3 id="prepare-motif-pairs-and-add-chip-seq-data">Prepare motif pairs and add ChIP-seq data<a class="anchor" aria-label="anchor" href="#prepare-motif-pairs-and-add-chip-seq-data"></a>
</h3>
<p>First, we need to build the pairs of motifs as candidates and add the
ChIP-seq data as shown above.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># load provided CTCF motifs</span></span>
<span><span class="va">motifs</span> <span class="op">&lt;-</span> <span class="va">motif.hg19.CTCF.chr22</span></span>
<span></span>
<span><span class="co"># use example ChIP-seq coverage file</span></span>
<span><span class="va">bigWigFile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"GM12878_Stat1.chr22_1-30000000.bigWig"</span>, </span>
<span>  package <span class="op">=</span> <span class="st">"sevenC"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># add ChIP-seq coverage</span></span>
<span><span class="va">motifs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/addCovToGR.html">addCovToGR</a></span><span class="op">(</span><span class="va">motifs</span>, <span class="va">bigWigFile</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># build motif pairs</span></span>
<span><span class="va">gi</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prepareCisPairs.html">prepareCisPairs</a></span><span class="op">(</span><span class="va">motifs</span>, maxDist <span class="op">=</span> <span class="fl">10</span><span class="op">^</span><span class="fl">6</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># add correaltion of ChIP-signal</span></span>
<span><span class="va">gi</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/addCovCor.html">addCovCor</a></span><span class="op">(</span><span class="va">gi</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="train-predictor-with-known-loops">Train predictor with known loops<a class="anchor" aria-label="anchor" href="#train-predictor-with-known-loops"></a>
</h3>
<p>We need to label true looping interactions by using experimental data
of chromatin interactions. Here, we use loops from high-resolution Hi-C
experiments in human GM12878 cells <span class="citation">(Rao et al.
2014)</span>. An example file with loops on chromosome 22 is provided
with the <em><a href="https://bioconductor.org/packages/3.22/sevenC" class="external-link">sevenC</a></em>
package and the function <code><a href="../reference/parseLoopsRao.html">parseLoopsRao()</a></code> reads loops in the
format provided by Rao et al. and returns a <code>GInteraction</code>
object.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># parse known loops</span></span>
<span><span class="va">knownLoopFile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, </span>
<span>  <span class="st">"GM12878_HiCCUPS.chr22_1-30000000.loop.txt"</span>, package <span class="op">=</span> <span class="st">"sevenC"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">knownLoops</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/parseLoopsRao.html">parseLoopsRao</a></span><span class="op">(</span><span class="va">knownLoopFile</span><span class="op">)</span></span></code></pre></div>
<p>We can add a new metadata column to the motif pairs <code>gi</code>,
indicating whether the pair is interacting in the experimental data
using the function <code><a href="../reference/addInteractionSupport.html">addInteractionSupport()</a></code>.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># add known loops</span></span>
<span><span class="va">gi</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/addInteractionSupport.html">addInteractionSupport</a></span><span class="op">(</span><span class="va">gi</span>, <span class="va">knownLoops</span><span class="op">)</span></span></code></pre></div>
<p>The experimental support is added as factor with levels
<code>"Loop"</code> and <code>"No loop"</code> as metadata column named
<code>loop</code>. The column name can be modified using the
<code>colname</code> argument.</p>
</div>
<div class="section level3">
<h3 id="train-logistic-regression-model">Train logistic regression model<a class="anchor" aria-label="anchor" href="#train-logistic-regression-model"></a>
</h3>
<p>We can use the R function <code><a href="https://rdrr.io/r/stats/glm.html" class="external-link">glm()</a></code> to fit a logistic
regression model in which the <code>loop</code> column is the dependent
variable and the ChIP-seq correlation, distance, and strand orientation
are the predictors.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/glm.html" class="external-link">glm</a></span><span class="op">(</span></span>
<span>  formula <span class="op">=</span> <span class="va">loop</span> <span class="op">~</span> <span class="va">cor_chip</span> <span class="op">+</span> <span class="va">dist</span> <span class="op">+</span> <span class="va">strandOrientation</span>, </span>
<span>  data <span class="op">=</span> <span class="fu">mcols</span><span class="op">(</span><span class="va">gi</span><span class="op">)</span>, </span>
<span>  family <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/family.html" class="external-link">binomial</a></span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="predict-loops-with-a-custom-model">Predict loops with a custom model<a class="anchor" aria-label="anchor" href="#predict-loops-with-a-custom-model"></a>
</h3>
<p>Now, we can use this model to add predicted looping
probabilities.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># add predict loops</span></span>
<span><span class="va">gi</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/predLoops.html">predLoops</a></span><span class="op">(</span></span>
<span>  <span class="va">gi</span>,</span>
<span>  formula <span class="op">=</span> <span class="va">loop</span> <span class="op">~</span> <span class="va">cor_chip</span> <span class="op">+</span> <span class="va">dist</span> <span class="op">+</span> <span class="va">strandOrientation</span>,</span>
<span>  betas <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/coef.html" class="external-link">coef</a></span><span class="op">(</span><span class="va">fit</span><span class="op">)</span>,</span>
<span>  cutoff <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>Here, we have to use the same formula as argument as in the model
fitting step above. The <code>betas</code> argument takes the
coefficients of the logistic regression model. Finally, the argument
<code>cutoff = NULL</code> ensures that no filtering is done and all
input candidates are reported. The prediction score is added as a new
metadata column to <code>gi</code>.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gi</span> </span></code></pre></div>
<pre><code><span><span class="co">## StrictGInteractions object with 26076 interactions and 8 metadata columns:</span></span>
<span><span class="co">##           seqnames1           ranges1     seqnames2           ranges2 |</span></span>
<span><span class="co">##               &lt;Rle&gt;         &lt;IRanges&gt;         &lt;Rle&gt;         &lt;IRanges&gt; |</span></span>
<span><span class="co">##       [1]     chr22 16186188-16186206 ---     chr22 16205307-16205325 |</span></span>
<span><span class="co">##       [2]     chr22 16186188-16186206 ---     chr22 16238548-16238566 |</span></span>
<span><span class="co">##       [3]     chr22 16186188-16186206 ---     chr22 16239188-16239206 |</span></span>
<span><span class="co">##       [4]     chr22 16186188-16186206 ---     chr22 16239827-16239845 |</span></span>
<span><span class="co">##       [5]     chr22 16186188-16186206 ---     chr22 16247246-16247264 |</span></span>
<span><span class="co">##       ...       ...               ... ...       ...               ... .</span></span>
<span><span class="co">##   [26072]     chr22 51110780-51110798 ---     chr22 51160028-51160046 |</span></span>
<span><span class="co">##   [26073]     chr22 51110780-51110798 ---     chr22 51172084-51172102 |</span></span>
<span><span class="co">##   [26074]     chr22 51130130-51130148 ---     chr22 51160028-51160046 |</span></span>
<span><span class="co">##   [26075]     chr22 51130130-51130148 ---     chr22 51172084-51172102 |</span></span>
<span><span class="co">##   [26076]     chr22 51160028-51160046 ---     chr22 51172084-51172102 |</span></span>
<span><span class="co">##                dist strandOrientation   score_1   score_2 score_min  cor_chip</span></span>
<span><span class="co">##           &lt;integer&gt;       &lt;character&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">##       [1]     19119           forward      6.05      5.72      5.72  0.339540</span></span>
<span><span class="co">##       [2]     52360           forward      6.05      5.92      5.92 -0.104692</span></span>
<span><span class="co">##       [3]     53000           forward      6.05      5.90      5.90  0.885073</span></span>
<span><span class="co">##       [4]     53639           forward      6.05      5.92      5.92 -0.170961</span></span>
<span><span class="co">##       [5]     61058           forward      6.05      5.92      5.92 -0.104692</span></span>
<span><span class="co">##       ...       ...               ...       ...       ...       ...       ...</span></span>
<span><span class="co">##   [26072]     49248           reverse      6.21      5.64      5.64        NA</span></span>
<span><span class="co">##   [26073]     61304           reverse      6.21      5.90      5.90        NA</span></span>
<span><span class="co">##   [26074]     29898           reverse      6.33      5.64      5.64        NA</span></span>
<span><span class="co">##   [26075]     41954           reverse      6.33      5.90      5.90        NA</span></span>
<span><span class="co">##   [26076]     12056           reverse      5.64      5.90      5.64        NA</span></span>
<span><span class="co">##               loop        pred</span></span>
<span><span class="co">##           &lt;factor&gt;   &lt;numeric&gt;</span></span>
<span><span class="co">##       [1]  No loop 0.002077368</span></span>
<span><span class="co">##       [2]  No loop 0.000422511</span></span>
<span><span class="co">##       [3]  No loop 0.012289515</span></span>
<span><span class="co">##       [4]  No loop 0.000335909</span></span>
<span><span class="co">##       [5]  No loop 0.000414237</span></span>
<span><span class="co">##       ...      ...         ...</span></span>
<span><span class="co">##   [26072]  No loop          NA</span></span>
<span><span class="co">##   [26073]  No loop          NA</span></span>
<span><span class="co">##   [26074]  No loop          NA</span></span>
<span><span class="co">##   [26075]  No loop          NA</span></span>
<span><span class="co">##   [26076]  No loop          NA</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   regions: 917 ranges and 2 metadata columns</span></span>
<span><span class="co">##   seqinfo: 93 sequences (1 circular) from hg19 genome</span></span></code></pre>
<p>As a very simple validation, we can now compare the prediction score
for looping and non-looping motif pairs using a boxplot.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/boxplot.html" class="external-link">boxplot</a></span><span class="op">(</span><span class="va">gi</span><span class="op">$</span><span class="va">pred</span> <span class="op">~</span> <span class="va">gi</span><span class="op">$</span><span class="va">loop</span>, </span>
<span>        ylab <span class="op">=</span> <span class="st">"Predicted interaction probability"</span><span class="op">)</span></span></code></pre></div>
<p><img src="sevenC_files/figure-html/unnamed-chunk-22-1.png" width="288"></p>
<p>The plot shows higher prediction scores for truly looping motif
pairs. However, this is an insufficient evaluation of prediction
performance, since the prediction score is evaluated on the same data as
it was trained. A more detailed evaluation of prediction performance
using cross-validation and different cell types is described in the 7C
paper <span class="citation">(Ibn-Salem and Andrade-Navarro
2018)</span>.</p>
</div>
<div class="section level3">
<h3 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h3>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.5.1 (2025-06-13)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">## Running under: Ubuntu 24.04.3 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] GenomicInteractions_1.43.1  sevenC_1.31.1              </span></span>
<span><span class="co">##  [3] InteractionSet_1.37.1       SummarizedExperiment_1.39.2</span></span>
<span><span class="co">##  [5] Biobase_2.69.1              MatrixGenerics_1.21.0      </span></span>
<span><span class="co">##  [7] matrixStats_1.5.0           GenomicRanges_1.61.5       </span></span>
<span><span class="co">##  [9] Seqinfo_0.99.2              IRanges_2.43.5             </span></span>
<span><span class="co">## [11] S4Vectors_0.47.4            BiocGenerics_0.55.3        </span></span>
<span><span class="co">## [13] generics_0.1.4              BiocStyle_2.37.1           </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] RColorBrewer_1.1-3       rstudioapi_0.17.1        jsonlite_2.0.0          </span></span>
<span><span class="co">##   [4] magrittr_2.0.4           GenomicFeatures_1.61.6   farver_2.1.2            </span></span>
<span><span class="co">##   [7] rmarkdown_2.30           fs_1.6.6                 BiocIO_1.19.0           </span></span>
<span><span class="co">##  [10] ragg_1.5.0               vctrs_0.6.5              memoise_2.0.1           </span></span>
<span><span class="co">##  [13] Rsamtools_2.25.3         RCurl_1.98-1.17          base64enc_0.1-3         </span></span>
<span><span class="co">##  [16] htmltools_0.5.8.1        S4Arrays_1.9.1           progress_1.2.3          </span></span>
<span><span class="co">##  [19] curl_7.0.0               SparseArray_1.9.1        Formula_1.2-5           </span></span>
<span><span class="co">##  [22] sass_0.4.10              bslib_0.9.0              htmlwidgets_1.6.4       </span></span>
<span><span class="co">##  [25] desc_1.4.3               Gviz_1.53.1              httr2_1.2.1             </span></span>
<span><span class="co">##  [28] cachem_1.1.0             GenomicAlignments_1.45.4 igraph_2.2.0            </span></span>
<span><span class="co">##  [31] lifecycle_1.0.4          pkgconfig_2.0.3          Matrix_1.7-4            </span></span>
<span><span class="co">##  [34] R6_2.6.1                 fastmap_1.2.0            digest_0.6.37           </span></span>
<span><span class="co">##  [37] colorspace_2.1-2         AnnotationDbi_1.71.1     textshaping_1.0.4       </span></span>
<span><span class="co">##  [40] Hmisc_5.2-4              RSQLite_2.4.3            filelock_1.0.3          </span></span>
<span><span class="co">##  [43] httr_1.4.7               abind_1.4-8              compiler_4.5.1          </span></span>
<span><span class="co">##  [46] bit64_4.6.0-1            htmlTable_2.4.3          S7_0.2.0                </span></span>
<span><span class="co">##  [49] backports_1.5.0          BiocParallel_1.43.4      DBI_1.2.3               </span></span>
<span><span class="co">##  [52] biomaRt_2.65.16          rappdirs_0.3.3           DelayedArray_0.35.3     </span></span>
<span><span class="co">##  [55] rjson_0.2.23             tools_4.5.1              foreign_0.8-90          </span></span>
<span><span class="co">##  [58] nnet_7.3-20              glue_1.8.0               restfulr_0.0.16         </span></span>
<span><span class="co">##  [61] grid_4.5.1               checkmate_2.3.3          cluster_2.1.8.1         </span></span>
<span><span class="co">##  [64] gtable_0.3.6             BSgenome_1.77.2          tzdb_0.5.0              </span></span>
<span><span class="co">##  [67] ensembldb_2.33.2         data.table_1.17.8        hms_1.1.3               </span></span>
<span><span class="co">##  [70] XVector_0.49.1           pillar_1.11.1            stringr_1.5.2           </span></span>
<span><span class="co">##  [73] vroom_1.6.6              dplyr_1.1.4              BiocFileCache_2.99.6    </span></span>
<span><span class="co">##  [76] lattice_0.22-7           deldir_2.0-4             rtracklayer_1.69.1      </span></span>
<span><span class="co">##  [79] bit_4.6.0                biovizBase_1.57.1        tidyselect_1.2.1        </span></span>
<span><span class="co">##  [82] Biostrings_2.77.2        knitr_1.50               gridExtra_2.3           </span></span>
<span><span class="co">##  [85] bookdown_0.45            ProtGenerics_1.41.0      xfun_0.53               </span></span>
<span><span class="co">##  [88] stringi_1.8.7            UCSC.utils_1.5.0         lazyeval_0.2.2          </span></span>
<span><span class="co">##  [91] yaml_2.3.10              boot_1.3-32              evaluate_1.0.5          </span></span>
<span><span class="co">##  [94] codetools_0.2-20         interp_1.1-6             tibble_3.3.0            </span></span>
<span><span class="co">##  [97] BiocManager_1.30.26      cli_3.6.5                rpart_4.1.24            </span></span>
<span><span class="co">## [100] systemfonts_1.3.1        jquerylib_0.1.4          dichromat_2.0-0.1       </span></span>
<span><span class="co">## [103] Rcpp_1.1.0               GenomeInfoDb_1.45.12     dbplyr_2.5.1            </span></span>
<span><span class="co">## [106] png_0.1-8                XML_3.99-0.19            parallel_4.5.1          </span></span>
<span><span class="co">## [109] readr_2.1.5              pkgdown_2.1.3.9000       ggplot2_4.0.0           </span></span>
<span><span class="co">## [112] blob_1.2.4               prettyunits_1.2.0        jpeg_0.1-11             </span></span>
<span><span class="co">## [115] latticeExtra_0.6-31      AnnotationFilter_1.33.0  bitops_1.0-9            </span></span>
<span><span class="co">## [118] VariantAnnotation_1.55.1 scales_1.4.0             purrr_1.1.0             </span></span>
<span><span class="co">## [121] crayon_1.5.3             rlang_1.1.6              KEGGREST_1.49.2</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Dunham2012" class="csl-entry">
Dunham, Ian, Anshul Kundaje, Shelley F Aldred, et al. 2012. <span>“<span class="nocase">An integrated encyclopedia of DNA elements in the human
genome.</span>”</span> <em>Nature</em> 489 (7414): 57–74. <a href="https://doi.org/10.1038/nature11247" class="external-link">https://doi.org/10.1038/nature11247</a>.
</div>
<div id="ref-Ibn-Salem2018" class="csl-entry">
Ibn-Salem, Jonas, and Miguel A. Andrade-Navarro. 2018. <span>“<span class="nocase">Computational Chromosome Conformation Capture by
Correlation of ChIP-seq at CTCF motifs</span>.”</span> <em>bioRxiv</em>,
February, 257584. <a href="https://doi.org/10.1101/257584" class="external-link">https://doi.org/10.1101/257584</a>.
</div>
<div id="ref-Khan2018" class="csl-entry">
Khan, Aziz, Oriol Fornes, Arnaud Stigliani, et al. 2018. <span>“<span class="nocase">JASPAR 2018: update of the open-access database of
transcription factor binding profiles and its web
framework</span>.”</span> <em>Nucleic Acids Research</em> 46 (D1):
D260–66. <a href="https://doi.org/10.1093/nar/gkx1126" class="external-link">https://doi.org/10.1093/nar/gkx1126</a>.
</div>
<div id="ref-Rao2014" class="csl-entry">
Rao, Suhas S P, Miriam H Huntley, Neva C Durand, et al. 2014.
<span>“<span class="nocase">A 3D map of the human genome at kilobase
resolution reveals principles of chromatin looping.</span>”</span>
<em>Cell</em> 159 (7): 1665–80. <a href="https://doi.org/10.1016/j.cell.2014.11.021" class="external-link">https://doi.org/10.1016/j.cell.2014.11.021</a>.
</div>
<div id="ref-Tang2015" class="csl-entry">
Tang, Zhonghui, Oscar Junhong Luo, Xingwang Li, et al. 2015.
<span>“<span class="nocase">CTCF-Mediated Human 3D Genome Architecture
Reveals Chromatin Topology for Transcription</span>.”</span>
<em>Cell</em>, 1–17. <a href="https://doi.org/10.1016/j.cell.2015.11.024" class="external-link">https://doi.org/10.1016/j.cell.2015.11.024</a>.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Jonas Ibn-Salem.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.9000.</p>
</div>

      </footer>
</div>






  </body>
</html>
